aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp

```diff
@@ -3046,48 +3052,48 @@ REGISTER_CUDA_DISPATCH(svd_stub, &svd_kernel)
 /*
   Solves the matrix equation A X = B
   X and B are n-by-nrhs matrices, A is represented using the LU factorization.
-  This is an in-place routine, content of `b` is overwritten.
+  This is an in-place routine, content of `B` is overwritten.
   This is a "looped" variant for calling single input MAGMA function on batched input.
 
   Args:
-  * `b` -  [in] the right hand side matrix B
-           [out] the solution matrix X
-  * `lu` - [in] the LU factorization of matrix A (see at::linalg_lu_factor)
+  * `LU` - [in] the LU factorization of matrix A (see at::linalg_lu_factor)
   * `pivots` - [in] the pivot indices (see at::linalg_lu_factor)
+  * `B` -  [in] the right hand side matrix B
+           [out] the solution matrix X
 
   For further details, please see the MAGMA documentation for magma_dgetrs_gpu.
 */
 template <typename scalar_t>
-static void apply_lu_solve_looped_magma(const Tensor& b, const Tensor& lu, const Tensor& pivots, TransposeType transpose) {
+static void apply_lu_solve_looped_magma(const Tensor& LU, const Tensor& pivots, const Tensor& B, TransposeType transpose) {
 #if !AT_MAGMA_ENABLED()
   TORCH_CHECK(
       false,
-      "Calling torch.lu_solve on a CUDA tensor requires compiling ",
+      "Calling linalg.lu_solve on a CUDA tensor requires compiling ",
       "PyTorch with MAGMA. Please rebuild with MAGMA.");
 #else
   auto trans = to_magma(transpose);
-  auto b_data = b.data_ptr<scalar_t>();
-  auto lu_data = lu.data_ptr<scalar_t>();
+  auto b_data = B.data_ptr<scalar_t>();
+  auto lu_data = LU.data_ptr<scalar_t>();
 
   // MAGMA requires pivots to be a CPU tensor
   Tensor pivots_cpu = pivots.cpu();
   auto pivots_data = pivots_cpu.data_ptr<magma_int_t>();
 
-  auto b_stride = matrixStride(b);
-  auto lu_stride = lu.dim() > 2 ? lu.stride(-3) : 0;
+  auto b_stride = matrixStride(B);
+  auto lu_stride = LU.dim() > 2 ? LU.stride(-3) : 0;
   auto pivots_stride = pivots_cpu.dim() > 1 ? pivots_cpu.stride(-2) : 0;
-  auto batch_size = batchCount(b);
+  auto batch_size = batchCount(B);
 
-  magma_int_t n = magma_int_cast(lu.size(-2), "n");
-  magma_int_t nrhs = magma_int_cast(b.size(-1), "nrhs");
+  magma_int_t n = magma_int_cast(LU.size(-2), "n");
+  magma_int_t nrhs = magma_int_cast(B.size(-1), "nrhs");
   auto leading_dimension = std::max<magma_int_t>(1, n);
 
-  // lu and pivots tensors can be broadcast to b
+  // LU and pivots tensors can be broadcasted to B
   // here we construct a helper indexing tensor to linearly index into lu and pivots
```

Comment:
Is `LU` broadcasting over `B` efficient? It is only `B` that is modified in-place, right?

Reply:
yes